unic-locale-impl/src/extensions/mod.rs

```diff
@@ -58,24 +57,30 @@ pub struct ExtensionsMap {
 }
 
 impl ExtensionsMap {
-    pub fn try_from_iter<'a>(
-        iter: &mut Peekable<impl Iterator<Item = &'a str>>,
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, ParserError> {
+        let mut iterator = bytes.split(|c| *c == b'-' || *c == b'_').peekable();
+        Self::try_from_iter(&mut iterator)
+    }
+
+    pub(crate) fn try_from_iter<'a>(
+        iter: &mut Peekable<impl Iterator<Item = &'a [u8]>>,
     ) -> Result<Self, ParserError> {
         let mut result = ExtensionsMap::default();
 
         let mut st = iter.next();
         while let Some(subtag) = st {
-            let subtag = subtag.to_ascii_lowercase();
+            if subtag.is_empty() {
+                break;
+            }
 
-            match subtag.as_str() {
-                "" => break,
-                "u" => {
+            match ExtensionType::from_byte(subtag[0]) {
```

Comment:
What guarantees that `subtag.len()` is one?

Reply:
nothing, but the line above guarantees it is at least length of 1 :)